scripting/get5/teamlogic.sp

```diff
@@ -74,21 +85,39 @@ public Action Command_JoinTeam(int client, const char[] command, int argc) {
       if (!g_CoachingEnabledCvar.BoolValue) {
         KickClient(client, "%t", "TeamIsFullInfoMessage");
       } else {
-        LogDebug("Forcing player %N to coach", client);
-        MoveClientToCoach(client);
-        Get5_Message(client, "%t", "MoveToCoachInfoMessage");
+        // Only attempt to move to coach if we are not full on coaches already.
+        if (GetTeamCoaches(correctTeam).Length <= g_CoachesPerTeam) {
+          LogDebug("Forcing player %N to coach", client);
+          MoveClientToCoach(client);
+          Get5_Message(client, "%t", "MoveToCoachInfoMessage");
+        } else {
+          KickClient(client, "%t", "TeamIsFullInfoMessage");
+        }
       }
     } else {
       LogDebug("Forcing player %N onto %d", client, csTeam);
       FakeClientCommand(client, "jointeam %d", csTeam);
     }
-
+    
+    CheckIfClientCoaching(client, correctTeam);
     return Plugin_Stop;
   }
 
   return Plugin_Stop;
 }
 
+public bool CheckIfClientCoaching(int client, MatchTeam team) {
+    // Force user to join the coach if specified by config or reconnect.
+    char clientAuth64[AUTH_LENGTH];
+    GetAuth(client, clientAuth64, AUTH_LENGTH);
```

Comment:
Move getAuth inside the "if coaching" check. No reason to fetch if it should not be evaluated.

Reply:
or just return false early at the beginning if g_CoachingEnabledCvar.BoolValue is false